### PR TITLE
Automated cherry pick of #65424: Fix scheduler config decoding

### PR DESCRIPTION
**Pull request by liggitt**
Cherry pick of #65424 on release-1.11.

#65424: Fix scheduler config decoding

### CODE REVIEW

**Comment by liggitt** [comment]
/sig scheduling
/kind bug
/priority critical-urgent
/status approved-for-milestone

@kubernetes/sig-scheduling-bugs

**Review by resouer — COMMENTED** [review]
This lgtm if we agree on https://github.com/kubernetes/kubernetes/pull/65415#discussion_r197670198

Otherwise, we should include all released fields in `TestCompatibility_v1_Scheduler... `

**Comment by nikhita** [comment]
/assign @foxish 
for cherry-pick approval

File: pkg/scheduler/algorithmprovider/defaults/compatibility_test.go [hunk 1]
```diff
@@ -698,6 +793,17 @@ func TestCompatibility_v1_Scheduler(t *testing.T) {
...
+				ExtenderConfigs: []schedulerapi.ExtenderConfig{{
+					URLPrefix:        "/prefix",
+					FilterVerb:       "filter",
+					PrioritizeVerb:   "prioritize",
+					Weight:           1,
+					BindVerb:         "bind", // 1.11 restored case-sensitivity, but allowed either "BindVerb" or "bindVerb"
+					EnableHTTPS:      true,
+					TLSConfig:        &restclient.TLSClientConfig{Insecure: true},
+					HTTPTimeout:      1,
+					NodeCacheCapable: true,
```
[the remark this answers is not in the document]
Comment (liggitt) [reply pkg/scheduler/algorithmprovider/defaults/compatibility_test.go]: done

**Pull request merged**